modules/auth.php

```diff
@@ -260,6 +262,94 @@ function handler_admin_authgroupesx($page, $action = 'list', $id = null)
         $table_editor->describe('last_used', 'dernière utilisation', true);
         $table_editor->apply($page, $action, $id);
     }
+
+    /** Handles the Discourse SSO authentication.
+     * https://meta.discourse.org/t/official-single-sign-on-for-discourse/13045
+     * Expects the Discourse domain as the last component of the URL path
+     * Expects the following GET parameters: sso, sig
+     * e.g. https://www.polytechnique.org/auth-discourse/forum.polytechnique.org?sso=...&sig=...
+     */
+    function handler_discourse($page, $domain = '')
+    {
+        global $globals;
+
+        // Load the key
+        if (!preg_match('/^[a-zA-Z0-9\-.:]+$/', $domain)) {
```

Comment:
I used it to test my Discourse instance running at ``localhost:8000`` but I will drop the ``:`` in the final patch.